tremor-cli/src/main.rs

```diff
@@ -107,8 +103,9 @@ async fn main() -> Result<()> {
 }
 
 async fn run(cli: Cli) -> Result<()> {
-    // Logging
-    if let Some(logger_config) = &cli.logger_config {
+    if cli.pluggable_logging {
```

Comment:
I'm not sure we want them to be exclusive, I can totally see a user wanting to have a rotating file along with ingesting the metrics to a pipeline, but I could make a use-case up that doesn't exit.

If they are exclusive, we definitely want to error when both are provided however otherwise users will have to deal with their logging config silently being ignored and that'd be bad :tm:

Looking at the code wouldn't it make more sense to allow `tremor` to be caused as an appended so the user can configure it? Having it forced to always be info for everything seems like a problem in usability, most of the time users will want to limit logs to `warning` or `error` to keep down noise. 

Looking at the other code, guardingthis is quite tricky as the functions and connectors still would be present, just not usable, the more I dig into it the more I get the impression this perhaps should be always on.